actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -542,7 +542,7 @@ def current_page?(options, check_parameters: false)
 
         return false unless request.get? || request.head?
 
-        check_parameters ||= !options.is_a?(String) && options.try(:delete, :check_parameters)
+        check_parameters ||= options.respond_to?(:to_hash) && options.try(:delete, :check_parameters)
```

Comment:
This still feels risky to me - would not going with an established pattern be better, e.g:

``` ruby
def current_page?(*args)
  options = args.extract_options!
  check_parameters = options.delete(:check_parameters)
  url_for_options = args.first || options

  # rest of implementation
end
```

If a model implements `to_hash` for some reason then it would still call `delete` on it.

@matthewd wdyt? (also wasn't my fault :wink:)